src/main/java/com/teamscale/upload/utils/OkHttpUtils.java

```diff
@@ -61,75 +73,130 @@ public static OkHttpClient createClient(boolean validateSsl, String trustStorePa
+			SSLContext sslContext = SSLContext.getInstance(PROTOCOL);
 
-			if (trustManagerFactory.getTrustManagers().length == 0) {
-				LogUtils.fail("No trust managers found. This is a bug. Please report it to CQSE.");
+			List<TrustManager> trustManagers = new ArrayList<>();
+			trustManagers.addAll(getJVMTrustManagers());
+			trustManagers.addAll(getOSTrustManagers());
+			if (trustStorePath != null) {
+				trustManagers.addAll(getExternalTrustManagers(trustStorePath, trustStorePassword));
 			}
 
-			applySslContextAndTrustManager(builder, sslContext, trustManagerFactory);
+			MultiTrustManager multiTrustManager = new MultiTrustManager(trustManagers);
+
+			sslContext.init(null, new TrustManager[]{multiTrustManager}, new SecureRandom());
+			builder.sslSocketFactory(sslContext.getSocketFactory(), multiTrustManager);
 		} catch (NoSuchAlgorithmException e) {
 			LogUtils.failWithStackTrace("Failed to instantiate an SSLContext or TrustManagerFactory."
 					+ "\nThis is a bug. Please report it to CQSE.", e);
-		} catch (KeyStoreException e) {
-			LogUtils.failWithStackTrace("Failed to initialize the TrustManagerFactory with the keystore."
-					+ "\nThis is a bug. Please report it to CQSE.", e);
 		} catch (KeyManagementException e) {
 			LogUtils.failWithStackTrace("Failed to initialize the SSLContext with the trust managers."
 					+ "\nThis is a bug. Please report it to CQSE.", e);
-		}
-	}
-
-	private static void applySslContextAndTrustManager(OkHttpClient.Builder builder, SSLContext sslContext,
-			TrustManagerFactory trustManagerFactory) {
-		try {
-			X509TrustManager x509TrustManager = (X509TrustManager) trustManagerFactory.getTrustManagers()[0];
-			builder.sslSocketFactory(sslContext.getSocketFactory(), x509TrustManager);
 		} catch (ClassCastException e) {
 			LogUtils.failWithStackTrace(
 					"Trust manager is not of X509 format." + "\nThis is a bug. Please report it to CQSE.", e);
 		}
 	}
 
-	private static KeyStore readKeyStore(String keystorePath, String keystorePassword) {
+	private static List<TrustManager> getExternalTrustManagers(String keystorePath, String keystorePassword) {
 		try (FileInputStream stream = new FileInputStream(keystorePath)) {
 			KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
 			keyStore.load(stream, keystorePassword.toCharArray());
-			return keyStore;
+
+			TrustManagerFactory trustManagerFactory = TrustManagerFactory
+					.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+			trustManagerFactory.init(keyStore);
+
+			List<TrustManager> trustManagers = List.of(trustManagerFactory.getTrustManagers());
+
+			if (trustManagers.size() == 0) {
+				LogUtils.fail("No custom trust managers found. This is a bug. Please report it to CQSE.");
+			}
+
+			return trustManagers;
+
+		} catch (NoSuchAlgorithmException e) {
+			LogUtils.failWithStackTrace("Failed to instantiate an SSLContext or TrustManagerFactory."
+					+ "\nThis is a bug. Please report it to CQSE.", e);
```

Comment:
Can we change the signature to `failWithStrace(Throwable e, String... lines)`?

Would highlight the importance of `e`, make the string-handling less awkward (`+`, `"\n"`), and allow us to use a constant for `"This is a bug ..."`